frontend/src/components/manageUsers/RoleChangeModal.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+import React, { RefObject } from "react";
+import {
+  Button,
+  ButtonGroup,
+  ModalFooter,
+  ModalRef,
+  ModalToggleButton,
+} from "@trussworks/react-uswds";
+
+import { SimplerModal } from "src/components/SimplerModal";
+
+export interface RoleChangeModalProps {
+  isSubmitting: boolean;
+  modalRef: RefObject<ModalRef | null>;
+  nextRoleName: string;
+  onConfirm: () => void;
+  onCancel: () => void;
+}
+
+export function RoleChangeModal({
+  isSubmitting,
+  modalRef,
+  nextRoleName,
+  onConfirm,
+  onCancel,
+}: RoleChangeModalProps) {
+  const t = useTranslations("ManageUsers.confirmationModal");
+  const modalTitle = t("header");
+  const modalDescription = `${t("description")} ${nextRoleName}?`;
+
+  return (
+    <SimplerModal
+      modalId="confirmation-role-change-modal"
+      modalRef={modalRef}
+      titleText={modalTitle}
+      className="text-wrap"
+    >
+      <p className="font-sans-2xs margin-y-4">{modalDescription}</p>
+      <ModalFooter>
+        <ButtonGroup>
+          <Button
+            type="button"
+            onClick={onConfirm}
+            disabled={isSubmitting}
+            aria-busy={isSubmitting}
+          >
+            {isSubmitting ? (
+              <span className="display-inline-flex flex-align-center">
```

Comment:
seems like we should have a shared component for this type of button state. Is this implementation borrowed from somewhere else?